tutorials/23_MultiHop_QA_With_Agents.ipynb

```diff
@@ -0,0 +1,5680 @@
+            "6ffba94cc23846209c4629cb10bce042",
+            "de52fc90bd3f4db0b540535246b33c4f",
+            "acd5844ab2d64071a1d59a1df7baafbd",
+            "744746530a4e401bb6e9e28d89e54eed",
+            "3bb3e922fc8c471a8c1aaaf5bbca2d92"
+          ]
+        },
+        "id": "HdS-satXbSsw",
+        "outputId": "751e30c8-50ae-48d3-8e34-df7023d1252a"
+      },
+      "outputs": [],
+      "source": [
+        "from haystack.utils import print_answers\n",
+        "\n",
+        "result = presidents_qa.run(\"Who was the 44th president of the USA\")\n",
+        "\n",
+        "print_answers(result, \"minimum\")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "bn83I_Dg8Q1f"
+      },
+      "source": [
+        "## Create an Agent with the `ExtractiveQAPipeline` as a `Tool`\n",
+        "### 1) To create the Agent, we will make use of an OpenAI model. So first, provide your OpenAI Key "
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "kBkMr86vvl9h",
+        "outputId": "8366a06d-9064-485e-aa88-c82859c68265"
+      },
+      "outputs": [],
+      "source": [
+        "from getpass import getpass\n",
+        "\n",
+        "api_key_prompt = \"Enter OpenAI API key:\" \n",
+        "api_key = getpass(api_key_prompt)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "qajsCzjx9C6j"
+      },
+      "source": [
+        "### 2) Create the Agent \n",
+        "By default, the `Agent` uses a `PromptNode` that makes use of the [\"zero-shot-react\" `PromptTemplate` ](https://github.com/deepset-ai/haystack/blob/444a3116c42d2c8852d27aa8093ac92c8e85ab88/haystack/nodes/prompt/prompt_node.py#L337). Here, let's define an `Agent` that uses the `text-davinci-003` model:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "uU20HAGXcf-q"
+      },
+      "outputs": [],
+      "source": [
+        "from haystack.agents import Agent\n",
+        "from haystack.nodes import PromptModel, PromptNode\n",
+        "\n",
+        "prompt_model = PromptModel(model_name_or_path=\"text-davinci-003\", api_key=api_key)\n",
+        "prompt_node = PromptNode(model_name_or_path=prompt_model, stop_words=[\"Observation:\"])\n",
+        "agent = Agent(prompt_node=prompt_node)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Ndeax5iGA1IQ"
+      },
+      "source": [
+        "### 3) Provide the Agent with a Tool\n",
+        "Next, let's add our `ExtractiveQAPipeline` into the Agents arsenal. The Agent will then have the ability to use this pipeline when it makes the decision that it would be useful.\n",
+        "\n",
+        "To do so, let's define a tool and make sure to give it a description:"
```

Comment:
Let's explain the importance of the description here. Something like:
"The exact wording of your description matters a lot here as the agent uses it to understand in which cases it should use this tool. If the agent fails to pick the right tool, adjusting the description might help"